gradient/commands/datasets.py

```diff
@@ -561,141 +561,139 @@ def update_status():
+                    # less than the part_minsize, AND we want to 1-index
+                    # our range to match what AWS expects for part
+                    # numbers
+                    for part in range(1, (size // part_minsize) + 2):
+                        presigned_url_res = api_client.post(
+                            url=mpu_url,
+                            json={
+                                'datasetId': dataset_id,
+                                'version': version,
+                                'calls': [{
+                                    'method': 'uploadPart',
+                                    'params': {
+                                        'Key': key,
+                                        'UploadId': mpu_data['UploadId'],
+                                        'PartNumber': part
+                                    }
+                                }]
+                            }
+                        )
+
+                        presigned_url = presigned_url_res.json()[0]['url']
+
+                        chunk = f.read(part_minsize)
+
+                        for attempt in range(0, 5):
+                            part_res = session.put(
+                                presigned_url,
+                                data=chunk,
+                                headers=headers,
+                                timeout=300)
+
+                            if part_res.status_code == 200:
+                                break
+
+                        if part_res.status_code != 200:
+                            # Why do we silence exceptions that get
+                            # explicitly raised? Mystery for the ages, but
+                            # there you have it I guess...
+                            print(f'\nUnable to complete upload of {path}')
+                            raise ApplicationError(
+                                f'Unable to complete upload of {path}')
+                        etag = part_res.headers['ETag'].replace('"', '')
+                        parts.append({'ETag': etag, 'PartNumber': part})
+                        # This is a pretty jank way to get about multipart
+                        # upload status updates, but we structure the Halo
+                        # spinner to report on the number of completed
+                        # tasks dispatched to the workers in the pool.
+                        # Since it's more of a PITA to properly distribute
+                        # this MPU among all workers than I really want to
+                        # deal with, that means we can't easily plug into
+                        # Halo for these updates. But we can print to
+                        # console! Which again, jank and noisy, but arguably
+                        # better than a task sitting forever, never either
+                        # completing or emitting an error message.
+                        print(
```

Comment:
Report every chunk

Reply:
This isn't too noisy now that we removed branch around it?

Reply:
No, I don't think the previous one was noisy enough honestly. It feels like nothing is happening. In general, we just need a better progress bar.